petab/v2/core.py

```diff
@@ -43,6 +49,43 @@
 ]
 
 
+def _is_finite_or_neg_inf(v: float, info: ValidationInfo) -> float:
```

Comment:
Move these methods to the lint module? I guess they might be used there too, or is all validation intended to take place in the constructor of these pydantic objects?

Reply:
That is still somewhat of an open question. I wouldn't want to duplicate the validation logic. However, the question is, which validations might have to be made optional at some point to work with future extensions. Probably too worry about that though.

So far, I would see everything that concerns the files and the presence of columns, as well as everything that cannot be validated inside a specific pydantic model to be in the lint module.

Another open question related to validation is whether it should still be possible to construct the pydantic objects from an invalid table.

Reply:
No strong opinion but I guess that could be useful for e.g. loading partial problems into Paul's PEtab GUI